app/models/piece.rb

```diff
@@ -2,5 +2,41 @@ class Piece < ActiveRecord::Base
   
   belongs_to :game
   belongs_to :player
+
+  def is_obstructed?(destination_x, destination_y)
+    location_x = self.x_position
+    location_y = self.y_position
+    if location_x == destination_x
+      # check for vertical obstruction
+      location_y > destination_y ? incrementer = -1 : incrementer = 1
+      position_y = location_y + incrementer
+      range = [position_y, destination_y].min...[position_y, destination_y].max
```

Comment:
there's actually a .minmax function that would be slightly more efficient, but this works too. 
```
min, max = [position_y, destination_y].minmax
range = min...max
```